### PR TITLE
Cache Docker build image on GitHub Docker Registry

### PR DESCRIPTION
**Pull request by arielshaqed**
Loosely based on [Caching Docker builds in GitHub Actions: Which approach is the fastest? 🤔 A
research](https://dev.to/dtinth/caching-docker-builds-in-github-actions-which-approach-is-the-fastest-a-research-18ei),
which indicates it should be the fastest.

### CODE REVIEW

**Comment by arielshaqed** [comment]
Shaves ~100s off Nessie time when everything seems ready.  Compare [this on `master` ](https://github.com/treeverse/lakeFS/runs/2703634805?check_suite_focus=true) without the patch (5m6s) with [this](https://github.com/treeverse/lakeFS/runs/2704258565?check_suite_focus=true) with the patch (3m24s).

If we can move more setup time into the Docker build we should be able to get more caching action, so that's probably next.

File: .github/workflows/nessie.yaml [hunk 1]
```diff
@@ -90,18 +90,47 @@ jobs:
...
+          if [[ -n $GITHUB_REF ]]; then ghr_tags=(${ghr_tags[@]} ${GITHUB_REF##*/}); fi
+          ghr_repo_tags=$(printf "%s " "${ghr_tags[@]/#/-t ${ghr}:}")
+          docker build . \
+              $ghr_repo_tags \
+              -t $ECR_REGISTRY/$ECR_REPOSITORY_LAKEFS:${{ steps.version.outputs.tag }} \
+              --build-arg VERSION=${{ steps.version.outputs.tag }} \
+              --cache-from=${ghr}
+          for ghr_tag in ${ghr_tags[0]}; do
+              docker push ${ghr}:${ghr_tag}
+          done
```
Comment (itaiad200) [inline .github/workflows/nessie.yaml:123]: Drop the array and make our life simpler!

**Review by itaiad200 — APPROVED** [review]
Single comment, approving but do fix it :)

**Comment by arielshaqed** [comment]
Thanks!  Still shaves ~50s off of the Spark client run.  Nessie itself remains slow, will next focus attention/caching there

**Pull request merged**